x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -94,6 +100,19 @@ func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sd
 		return nil, fmt.Errorf("Invalid withdraw Amount")
 	}
 
+	poolTotalLiquidity, err := k.GetTotalPoolLiquidity(ctx, clPool.GetId())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(poolTotalLiquidity) < 2 {
+		return nil, fmt.Errorf("pool doesnot have 2 tokens")
+	}
+
+	if poolTotalLiquidity[0].Amount.LTE(withdrawAmount.RoundInt()) || poolTotalLiquidity[1].Amount.LTE(withdrawAmount.RoundInt()) {
```

Comment:
I think more importantly we need to check that a positon has enough liquidity to withdraw from. If a position has enough, then the pool will have enough for sure as well